### PR TITLE
feat(rollup-relayer): auto finalize batch when timeout in test env

### PR DESCRIPTION
**Pull request by colinlyguo**
### Purpose or design rationale of this PR

This PR adds a feature to auto-finalize batch when timeout in test env (e.g., Sepolia).
The change is compatible with Mainnet env.

### PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [x] feat: A new feature

### Deployment tag versioning

Has `tag` in `common/version.go` been updated or have you added `bump-version` label to this PR?

- [x] Yes

### Breaking change label

Does this PR have the `breaking-change` label?

- [x] No, this PR is not a breaking change

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205858160020486

### CODE REVIEW

File: rollup/internal/controller/relayer/l2_relayer.go [hunk 1]
```diff
@@ -482,71 +523,61 @@ func (r *Layer2Relayer) ProcessCommittedBatches() {
...
 		}
-
-		txID := hash + "-finalize"
-		// add suffix `-finalize` to avoid duplication with commit tx in unit tests
-		txHash, err := r.finalizeSender.SendTransaction(txID, &r.cfg.RollupContractAddress, big.NewInt(0), data, 0)
-		finalizeTxHash := &txHash
+	} else {
+		var err error
+		txCalldata, err = r.l1RollupABI.Pack(
+			"finalizeBatch",
```
Comment (zimpha) [inline rollup/internal/controller/relayer/l2_relayer.go:533]: You need to update abi to use `finalizeBatch`.

Comment (colinlyguo) [reply rollup/internal/controller/relayer/l2_relayer.go:533]: Yes. Updated in [fb96ec3](https://github.com/scroll-tech/scroll/pull/1004/commits/fb96ec39d1c24766d40d5003ea9c04ae6400e2fe) and also added a unit test.

File: rollup/conf/config.json [hunk 2]
```diff
@@ -59,7 +58,8 @@
         "try_times": 5,
         "base_url": "http://localhost:8750"
       },
-      "finalize_batch_interval_sec": 0,
+      "is_test_env": true,
```
Comment (georgehao) [inline rollup/conf/config.json]: How about change it to `environment=mainnet|sepolia|staging`, and a common method check finalize batch timeout

Comment (colinlyguo) [reply rollup/conf/config.json]: If so, Sepolia would enable "timeout finalize" by default. Is it OK to do this? and btw, `mainnet|sepolia|staging` can be obtained by chain ID.

**Review by zimpha — APPROVED** [review]

File: rollup/internal/controller/relayer/l2_relayer.go [hunk 3]
```diff
@@ -423,56 +428,90 @@ func (r *Layer2Relayer) ProcessCommittedBatches() {
...
 
 	batch := batches[0]
-	hash := batch.Hash
 	status := types.ProvingStatus(batch.ProvingStatus)
 	switch status {
 	case types.ProvingTaskUnassigned, types.ProvingTaskAssigned:
-		// The proof for this block is not ready yet.
-		return
+		now := time.Now()
+		elapsedTime := now.Sub(batch.CreatedAt)
```
Comment (0xmountaintop) [inline rollup/internal/controller/relayer/l2_relayer.go]:
```suggestion
		elapsedTime := now.Sub(batch.CommittedAt)
```

Comment (colinlyguo) [reply rollup/internal/controller/relayer/l2_relayer.go]: oh, this field is not used...

**Review by 0xmountaintop — APPROVED** [review]

File: rollup/internal/controller/relayer/l2_relayer.go [hunk 4]
```diff
@@ -423,107 +428,29 @@ func (r *Layer2Relayer) ProcessCommittedBatches() {
...
-	hash := batch.Hash
 	status := types.ProvingStatus(batch.ProvingStatus)
 	switch status {
 	case types.ProvingTaskUnassigned, types.ProvingTaskAssigned:
-		// The proof for this block is not ready yet.
-		return
-	case types.ProvingTaskVerified:
-		log.Info("Start to roll up zk proof", "hash", hash)
-		r.metrics.rollupL2RelayerProcessCommittedBatchesFinalizedTotal.Inc()
+		now := time.Now()
```
Comment (georgehao) [inline rollup/internal/controller/relayer/l2_relayer.go]: need use `utils.NowUTC()`. because the time stored in db is utc time.

Comment (colinlyguo) [reply rollup/internal/controller/relayer/l2_relayer.go]: changed in [9a6ac09](https://github.com/scroll-tech/scroll/pull/1004/commits/9a6ac09e79b292d16cc25f3316ed2cdaa5399a12)

**Review by georgehao — APPROVED** [review]

**Pull request merged**